src/rpc/core_rpc_server.cpp

```diff
@@ -1509,42 +1521,36 @@ namespace cryptonote
         m_bootstrap_height_check_time = current_time;
       }
 
-      uint64_t top_height;
-      crypto::hash top_hash;
-      m_core.get_blockchain_top(top_height, top_hash);
-      ++top_height; // turn top block height into blockchain height
+      uint64_t bootstrap_daemon_height;
+      if (!m_bootstrap_daemon->get_height(&bootstrap_daemon_height))
+      {
+        return false;
+      }
 
-      // query bootstrap daemon's height
-      cryptonote::COMMAND_RPC_GET_HEIGHT::request getheight_req;
-      cryptonote::COMMAND_RPC_GET_HEIGHT::response getheight_res;
-      bool ok = epee::net_utils::invoke_http_json("/getheight", getheight_req, getheight_res, m_http_client);
-      ok = ok && getheight_res.status == CORE_RPC_STATUS_OK;
+      uint64_t target_height = m_core.get_target_blockchain_height();
+      if (bootstrap_daemon_height < target_height)
+      { 
+        return m_bootstrap_daemon->handle_result(false);
+      }
 
-      m_should_use_bootstrap_daemon = ok && top_height + 10 < getheight_res.height;
-      MINFO((m_should_use_bootstrap_daemon ? "Using" : "Not using") << " the bootstrap daemon (our height: " << top_height << ", bootstrap daemon's height: " << (ok ? getheight_res.height : 0) << ")");
+      uint64_t top_height = m_core.get_current_blockchain_height();
```

Comment:
This is now aliasing `target_height`, but its not obvious why ... was this some technique to prevent further modifications below? Why not declare `top_height` on line 1530 and drop `target_height`? This would also elminate one additional `m_core.get_current_blockchain_height()` call which has locks, etc., and isn't cheap.

Reply:
Nope, i changed the logic.

Now i want to skip public nodes that are not synced with the network (having `height` less than known `target_height`).